Review the 2 property rendering paths

AnthonySteele opened the pull request:
They differ in the checks, for no good reason.
Bring the safety of each to the other. Use common code.

Commit:
Review the 2 property rendering paths
Bring the safety of each to the other

Commit:
Common code

Commit:
The common code existed already

jplebre commented on src/NLog.StructuredLogging.Json/Helpers/Mapper.cs:
```diff
@@ -70,20 +70,28 @@ public static void HarvestToDictionary(IDictionary source, IDictionary<string, o
         }
 
-        private static void HarvestToDictionary(string key, object value, IDictionary<string, object> result, string keyPrefixWhenCollision)
+        private static void HarvestToDictionary(IDictionary<string, object> dest, string key, object value, string keyPrefixWhenCollision)
```
result above, destination here?
Naming is a bit confusing

AnthonySteele replied:
I changed `result` -> `dest` as I thought that was a better name. And dest as the first param is possibly more conventional. Will change it in the other place as well.

jplebre reviewed (commented):
Test for the exception throwing on the flattenedlayout?

You mentioned you were removing common code but the duplication still there?
Not fussed about optimizing for a single repetition, just wondering if that's what you intended.

other than that LGTM

Commit:
Rename result -> dest in a few more places

AnthonySteele commented:
There is still some duplication, yes. there are 2 Layout classes (see https://github.com/justeat/NLog.StructuredLogging.Json/issues/34) and the `try ... Render .. catch` can't be common between them as the types differ. But they both now have a `AddRenderedValue` method to make it clear that they're similar.

The code to put items in the dictionary with optional prefix was the common part (and already existed in the `Mapper`.
I'll be happy with this PR if the 2 Layouts are robust as each other.

Commit:
Test with a failing layout in a FlattenedJsonLayout

AnthonySteele commented:
Test added!

Commit:
More tests - property key duplicates, not just property value render failures

Commit:
This happens multiple times in tests
Not sure if it happens in real use?
Should only affect setup phase, so perf impact

jplebre commented:
looks good, will read it again tomorrow and merge it

jplebre commented on src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs:
```diff
@@ -0,0 +1,94 @@
+                "\"prop1\":\"value1\",\"prop2\":\"2\"}"));
+        }
+
+        private void GivenLoggingIsConfiguredForTest(Target target)
```
Should this and subsequent methods be moved to the `SimpleConfigurator`?

Don't mind either way

AnthonySteele replied:
SimpleConfigurator is part of NLog, we don't change it.
